server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4796,6 +4815,40 @@ private boolean localDirectoryContains(Directory localDirectory, String file, lo
         return false;
     }
 
+    private void beforeSegmentDownload(RemoteRefreshSegmentTracker downloadStatsTracker, long incomingFileSize) {
+        if (!indexSettings.isRemoteStoreEnabled()) {
+            return;
+        }
+        downloadStatsTracker.incrementTotalDownloadsStarted();
+        downloadStatsTracker.addDownloadBytesStarted(incomingFileSize);
+    }
+
+    private void afterSegmentDownloadCompleted(
+        RemoteRefreshSegmentTracker downloadStatsTracker,
+        long downloadedFileSize,
+        long startTimeInNs
+    ) {
+        if (!indexSettings.isRemoteStoreEnabled()) {
+            return;
```

Comment:
This is not required as the concerned code runs for remote enabled indexes only.

Reply:
Already addressed above